test/e2e/osa_test.go

```diff
@@ -83,4 +83,24 @@ var _ = Describe("Openshift on Azure e2e tests", func() {
 		})
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	It("should run the correct image", func() {
+		pods, err := c.kc.CoreV1().Pods("").List(metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		// e2e check should ensure that no reg-aws images are running on box
+		for _, pod := range pods.Items {
+			for _, cntr := range pod.Spec.Containers {
+				Expect(strings.Contains(cntr.Image, "registry.reg-aws")).ToNot(BeTrue())
+			}
+		}
+
+		// Check all Configmaps for image format matches distribution
+		// format: registry.access.redhat.com/openshift3/ose-${component}:${version}
+		maps, err := c.kc.CoreV1().ConfigMaps("openshift-node").List(metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		for _, cm := range maps.Items {
+			Expect(strings.Contains(cm.Data["node-config.yaml"], "format: "+c.cs.Config.Images.Format)).To(BeTrue())
```

Comment:
As this stands I think this reduces to a unit test.
I think you should be determining the cluster OS (centos or rhel) by inspecting the Node object, then calculating what you expect the imageformat to be based on that, then testing it

Reply:
Updated to check master-00000's os type and then test the image formats.